src/site/topics/objects-review/objects-review.rst

```diff
@@ -2,314 +2,374 @@
+    *necessary*, they are very powerful.
+
+
+Accessors
+^^^^^^^^^
+
+* Below are the accessor/getter methods for the fields
+* All these methods do are return their respective values
+* They are ``public`` as they should be accessible outside the class
+* Note, however, that there are no methods to set any of the field values
+
+    * The ``Friend`` is immutable --- can access data, but cannot change it
+
+
+.. literalinclude:: /../main/java/Friend.java
+    :language: java
     :linenos:
+    :lineno-start: 28
+    :lines: 28-38
+
 
-        /**
-         * Update the email address. Will only work if the email address
-         * contains the @ symbol for simple data correctness check
-         *
-         * @param email     new email address
-         */
-        public void setEmail(String email) {
-            if (email.contains("@")) {
-                this.email = email;
-            }
-        }
 
 toString
 ^^^^^^^^
 
-* You may remember that ``__repr__`` in python was a special method that would return a string version of the object
-    * And, if you called ``print(some_object)``, the ``__repr__`` would automatically get called
+* In Python, for creating a string representation of an object, the ``__repr__`` magic method was used
+
+    * If one called ``print(some_object)``, the ``__repr__`` would automatically get called
+
+
+* All classes *inherited* a ``__repr__`` for free, but the default behaviors was not all too helpful
 
-* When we wrote our own classes, we ended up overriding the ``__repr__`` we got for free
-    * The default one simply prints out the object name and a memory address, which wasn't very helpful
     * ``<__main__.Friend object at 0x7f130d9c52e0>``
-    * *Technically*, the behaviour we get for free is inherited from the closest "super" class
-        * We will learn more about this later
+    * The inherited one simply provides the object name and a memory address
 
-* An example of a ``__repr__`` for our ``Friend`` class might look something like this
-* We just print out the name of the ``Friend`` and their email
+        * Inheritance is a topic discussed later in the course
 
-**Python**
+
+* If one wanted to change this behaviour, they would override the default ``__repr__``
+* An example of a ``__repr__`` for the ``Friend`` class in Python is below
 
 .. code-block:: python
     :linenos:
 
-    # Python --- __repr__
     def __repr__(self):
-        return self._first_name + " " + self._last_name \
-               + ":\t" + self._email
+        return f"Friend({self.first_name}, {self.last_name}, {self.email})"
 
+* An f-string was used in the above example, but string concatenation could have been used
+
+    * ``"Friend(" + self.first_name + ", " + self.last_name + ", " + self.email + ")"``
+
+
+* The same principal exists in Java, but the method is called ``toString``
+* The inherited behaviour is a little different --- it returns a string of the class name and a *hash code*
```

Comment:
Specifically, it returns `Object`'s hash code. It's the address of the instance in the jvm (more or less).